src/CommonLib/AdaptiveTimeout.cs

```diff
@@ -0,0 +1,208 @@
+    public async Task<NetAPIResult<T>> ExecuteNetAPIWithTimeout<T>(Func<CancellationToken, NetAPIResult<T>> func, CancellationToken parentToken = default) {
+        var result = await Timeout.ExecuteNetAPIWithTimeout(GetAdaptiveTimeout(), (timeoutToken) => _sampler.SampleExecutionTime(() => func(timeoutToken)), parentToken);
+        TimeSpikeSafetyValve(result.IsSuccess);
+        return result;
+    }
+
+    /// <summary>
+    /// Ignores the result of a function if it runs longer than a budgeted time, unblocking the caller.
+    /// Logs aggregate execution time data.
+    /// Manages its own timeout.
+    /// A cancellation token is passed to the executing function so it may exit cleanly if timeout is reached.
+    /// Please don't wrap a cached function in this timeout if adaptive timeouts enabled, normal distributions are better.
+    /// DO NOT use a single AdaptiveTimeout for multiple functions.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="func"></param>
+    /// <param name="parentToken"></param>
+    /// <returns>Returns a Fail result if a task runs longer than its budgeted time.</returns>
+    public async Task<SharpHoundRPC.Result<T>> ExecuteRPCWithTimeout<T>(Func<CancellationToken, SharpHoundRPC.Result<T>> func, CancellationToken parentToken = default) {
+        var result = await Timeout.ExecuteRPCWithTimeout(GetAdaptiveTimeout(), (timeoutToken) => _sampler.SampleExecutionTime(() => func(timeoutToken)), parentToken);
+        TimeSpikeSafetyValve(result.IsSuccess);
+        return result;
+    }
+
+    /// <summary>
+    /// Ignores the result of a function if it runs longer than a budgeted time, unblocking the caller.
+    /// Logs aggregate execution time data.
+    /// Manages its own timeout.
+    /// A cancellation token is passed to the executing function so it may exit cleanly if timeout is reached.
+    /// Please don't wrap a cached function in this timeout if adaptive timeouts enabled, normal distributions are better.
+    /// DO NOT use a single AdaptiveTimeout for multiple functions.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="func"></param>
+    /// <param name="parentToken"></param>
+    /// <returns>Returns a Fail result if a task runs longer than its budgeted time.</returns>
+    public async Task<SharpHoundRPC.Result<T>> ExecuteRPCWithTimeout<T>(Func<CancellationToken, Task<SharpHoundRPC.Result<T>>> func, CancellationToken parentToken = default) {
+        var result = await Timeout.ExecuteRPCWithTimeout(GetAdaptiveTimeout(), (timeoutToken) => _sampler.SampleExecutionTime(() => func(timeoutToken)), parentToken);
+        TimeSpikeSafetyValve(result.IsSuccess);
+        return result;
+    }
+
+    public void Dispose() {
+        _sampler.Dispose();
+    }
+
+    // Within 5 standard deviations will have a conservative lower bound of catching 98% of executions (1 - 1/(k^2 / 2)),
```

Comment:
I do not see this " / 2" in the Tchebysheffs rule. Is there something else here going on?